api.go

```diff
@@ -651,11 +651,11 @@ func (c Client) newRequest(method string, metadata requestMetadata) (req *http.R
 		return req, nil
 	} // Sign the request for all authenticated requests.
 
-	if c.signature.isV2() {
+	switch {
+	case c.signature.isV2():
 		// Add signature version '2' authorization header.
 		req = s3signer.SignV2(*req, c.accessKeyID, c.secretAccessKey)
-	} else if c.signature.isV4() || c.signature.isStreamingV4() &&
-		method != "PUT" {
+	case c.signature.isV4():
```

Comment:
@krishnasrinivas the check is still required since the client's signature type is set before we know if the upload is going to be a single PUT object or a multipart upload. Streaming signature applies only to `http.MethodPUT`. API calls corresponding to get incomplete uploads, initiate new multipart, etc should use signature v4. Having said that, you could see if it can be refactored to be obvious to the reader and maintainable in the long run.

The current PR fails in `go test -v -run=TestPutObjectStreaming` when we make the following change. I have made the size of object to be uploaded to `64MB + 1` bytes.
```
diff --git a/api_functional_v4_test.go b/api_functional_v4_test.go
index dacbba8..c7c4a00 100644
--- a/api_functional_v4_test.go
+++ b/api_functional_v4_test.go
@@ -392,7 +392,7 @@ func TestPutObjectStreaming(t *testing.T) {
        }
 
        // Upload an object.
-       sizes := []int64{0, 64*1024 - 1, 64 * 1024}
+       sizes := []int64{0, 64*1024 - 1, 64*1024*1024 + 1}
        objectName := "test-object"
        for i, size := range sizes {
                data := bytes.Repeat([]byte("a"), int(size))
```